sql: add deprecation notice when using EXPORT

jayshrivastava opened the pull request:
### sql: add deprecation notice when using EXPORT #111415

This change adds a notice to EXPORT which says "EXPORT is not the recommended way to move data out of CockroachDB and may be deprecated in the future. Please consider exporting data with changefeeds instead:
https://www.cockroachlabs.com/docs/stable/export-data-with-changefeeds"

Closes: https://github.com/cockroachdb/cockroach/issues/111363

---

### backupccl: start with empty notice buffer in datadriven tests

Before this change, notices from previous commands (ex. EXPORT)
would stay in the buffer when running the IMPORT command (and
a few other commands).

This change updates the commands affected by this issue to clear
the notice buffer before starting.

Release note: None
Epic: None

jayshrivastava committed:
sql: add deprecation notice when using EXPORT

This change adds a notice to EXPORT which says "EXPORT is not the recommended
way to move data out of CockroachDB and may be deprecated in the future. Please
consider exporting data with changefeeds instead:
https://www.cockroachlabs.com/docs/stable/export-data-with-changefeeds"

Closes: https://github.com/cockroachdb/cockroach/issues/111363

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111415)
<!-- Reviewable:end -->

jayshrivastava committed:
backupccl: start with empty notice buffer in datadriven tests

Before this change, notices from previous commands (ex. EXPORT)
would stay in the buffer when running the IMPORT command (and
a few other commands).

This change updates the commands affected by this issue to clear
the notice buffer before starting.

Release note: None
Epic: None

jayshrivastava commented:
bors r+
